### ISSUE TITLE
tf.load_op_library behaves different in version 1.12 vs 1.11

### ISSUE THREAD

**Issue by christiantinauer** [tensorflow/tensorflow#23737]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): -
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Arch Linux
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: -
- TensorFlow installed from (source or binary): pip binary
- TensorFlow version (use command below): 1.12.0
- Python version: 3.6.5
- Bazel version (if compiling from source): -
- GCC/Compiler version (if compiling from source): gcc 6.5 for custom ops
- CUDA/cuDNN version: 9.0/7.3.1
- GPU model and memory: Titan X with 10GB

**Describe the current behavior**
When I use tf.load_op_library to load my library with two ops, only one of them appears in the loaded module.

**Describe the expected behavior**
Both ops should appear in the loaded module. Works in version 1.11.

**Comment by drpngx** [comment]
Do you have them both in the same so file? What are they called?

**Comment by christiantinauer** [comment]
They both should end up in the same shared object. The build script:

```
TF_CFLAGS=( $(python -c 'import tensorflow as tf; print(" ".join(tf.sysconfig.get_compile_flags()))') )
TF_LFLAGS=( $(python -c 'import tensorflow as tf; print(" ".join(tf.sysconfig.get_link_flags()))') )

nvcc \
  -DNDEBUG \
  -ccbin=g++-6 \
  -std=c++11 -c -o extract_volume_patches_op_gpu.cu.o \
  ../core/kernels/extract_volume_patches_op_gpu.cu.cc \
  ${TF_CFLAGS[@]} -D GOOGLE_CUDA=1 -x cu -Xcompiler -fPIC

nvcc \
  -DNDEBUG \
  -ccbin=g++-6 \
  -std=c++11 -c -o integrate_volume_patches_op_gpu.cu.o \
  ../core/kernels/integrate_volume_patches_op_gpu.cu.cc \
  ${TF_CFLAGS[@]} -D GOOGLE_CUDA=1 -x cu -Xcompiler -fPIC

g++-6 \
  -std=c++11 -shared -o volume_patches.so \
  ../core/ops/extract_volume_patches.cc \
  ../core/ops/integrate_volume_patches.cc \
  ../core/kernels/extract_volume_patches_op.cc \
  ../core/kernels/integrate_volume_patches_op.cc \
  extract_volume_patches_op_gpu.cu.o \
  integrate_volume_patches_op_gpu.cu.o \
  ${TF_CFLAGS[@]} -fPIC -lcudart ${TF_LFLAGS[@]} \
  -L /opt/cuda/lib64/
```

**Comment by drpngx** [comment]
I think I saw some similar problem recently. It's not clear what is going wrong. Did you build tensorflow from source? If not, could you try that?

@allenlavoie may have more ideas.

**Comment by allenlavoie** [comment]
Does it reproduce in a simple case, like doubling up the registration in [one of our custom op tests](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/kernel_tests/ackermann_op.cc)?

@annarev could this be a side effect of the API generation changes? 

CC @gunan @yifeif who are interested in custom ops

**Comment by annarev** [comment]
I can't think of recent api generation patches that could cause it. Api generation should not impact custom ops.

@christiantinauer What is the name of the op that doesn't load? I see tf.extract_volume_patches in 1.12 but not in 1.11:
https://www.tensorflow.org/api_docs/python/tf/extract_volume_patches

Could there be a possible name conflict:
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/framework/load_library.py#L47

**Comment by christiantinauer** [comment]
@annarev, @hsgkim actually it is a naming conflict as tf.extract_volume_patches is not loaded in version 1.12. Unfortunately the new op is not mentioned in the release notes (I checked before upgrading).

The implementation in 1.12 is nearly the same as mine. Nevertheless I also implemented the gradient for the op (integrate_volume_patches). If there is interest in the implementation I can share.

Thanks for the support.
